sys-block/gpart/patches/gpart-0.3.patchset

```diff
@@ -0,0 +1,91 @@
+From 83e2443543ac0162726c44de94fcbc35fa03eb01 Mon Sep 17 00:00:00 2001
+From: localanu <localanu@gmail.com>
+Date: Thu, 4 Jan 2018 07:01:46 +0000
+Subject: Try to add support to haiku
+
+
+diff --git a/autogen.sh b/autogen.sh
+old mode 100644
+new mode 100755
+diff --git a/src/disku.c b/src/disku.c
+index d50cebf..106f2ff 100644
+--- a/src/disku.c
++++ b/src/disku.c
+@@ -33,6 +33,9 @@
+ #include <sys/disk.h>
+ #endif
+ 
++#if defined(__HAIKU__)
++#include <Drivers.h>
++
+ #include <unistd.h>
+ 
+ static void geometry_from_num_sectors(struct disk_geom *g, uint64_t nsects)
+@@ -95,8 +98,21 @@ static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
+ 	g.d_nsecs = o / u;
+ 	g.d_c = g.d_nsecs / g.d_h / g.d_s;
+ }
++
++
++#elif defined(__HAIKU__)
++static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
++{
++	device_geometry dg;
++	ioctl(d->d_fd, B_GET_GEOMETRY, & dg);
++	g -> d_h = dg.head_count;
++	g -> d_c = dg.cylinder_count;
++	g -> d_s = dg.sectors_per_track * dg.cylinder_count;
++}
++#endif
++
+ #else
+-#error Only Linux and FreeBSD supported
++#error Only Linux, FreeBSD, and Haiku supported
+ #endif
+ 
+ /*
+diff --git a/src/gm_btrfs.c b/src/gm_btrfs.c
+index cdf0a36..155bb1d 100644
+--- a/src/gm_btrfs.c
++++ b/src/gm_btrfs.c
+@@ -12,7 +12,7 @@
+  * Created:   20.11.2015 <mwilck@arcor.de>
+  *
+  */
+-
++#define _BSD_SOURCE
```

Comment:
There should be no need for this. Your recipe already contains `-D_BSD_SOURCE` in `CFLAGS`

```diff
-+#define _BSD_SOURCE

Reply:
Need,error in building

Reply:
gm_bsddl.c:40:22: note: in expansion of macro 'le32'
  if ((dl->d_magic == le32(DISKMAGIC)) && (dl->d_magic2 == le32(DISKMAGIC))) {
                      ^
gpart.h:29:17: warning: implicit declaration of function 'htole16' [-Wimplicit-function-declaration]
 #define le16(x) htole16(x)

Reply:
and another gpart.h error

Reply:
It shouldn't. If there's any error, you should edit the `Makefile` to make sure CFLAGS is used when compiling this file

Reply:
wait,make file generated by autotools,not directly wirte